packages/mask/src/plugins/FindTruman/SNSAdaptor/index.tsx

```diff
@@ -30,7 +30,7 @@ const useStyles = makeStyles()((theme) => {
  */
 const isFindTrumanURL = (x: string): boolean =>
     // eslint-disable-next-line
-    /^https:\/\/findtruman.io\/#\/(findtruman\/stories\/[a-zA-Z0-9]+(\/|\/(puzzles|polls|puzzle_result|poll_result)\/[a-zA-Z0-9]+\/?)?|encryption\?payload=.+)$/i.test(
+    /^https:\/\/findtruman.io\/#\/(findtruman\/stories\/[a-zA-Z0-9]+(\/|\/(puzzles|polls|puzzle_result|poll_result)\/[a-zA-Z0-9]+\/?)?|encryption\?clueId=[a-zA-Z0-9]+)$/i.test(
```

Comment:
the function can simplify to multiple parts

1. check starts with is `https://findtruman.io`
2. use `URL` parse the string get `hash` part
3. use regexp pattern parse `hash` string

e.g:

```ts
if (!input.startsWith('https://findtruman.io')) {
    return false
}
const { pathname, hash } = new URL(input)
if (pathname !== '/') {
    return false
}
if (hash.startsWith('#/encryption')) {
    // your test regexp pattern.
}
if (hash.startsWith('#/findtruman/stories')) {
    // your test regexp pattern.
}
return false
```

see https://mdn.io/url-api